### PR TITLE
BUG: Escape strings for latex output

### LINKED ISSUE #5297: special Latex characters are not escaped in some table cells with Summary.as.latex()
With statsmodels 0.9.0:

Modifying the example from #2862:

```python
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf

dat = sm.datasets.get_rdataset("Guerry", "HistData").data

# Rename 'Lottery' column to 'Lottery_'
dat.rename(columns = {'Lottery':'Lottery_'}, inplace = True)

results = smf.ols('Lottery_ ~ Crime_prop + np.log(Pop1831)', data=dat).fit()

print(results.summary().as_latex())
```

The printed Latex successfully escapes the underscore in `Crime_prop`:

`\textbf{Crime\_prop}     &       0.0029  &        0.001     &     3.887  &         0.000        &        0.001    &        0.004     \\`

 but does not escape the underscore in Lottery_:

`\textbf{Dep. Variable:}    &     Lottery_     & \textbf{  R-squared:         } &     0.351   \\`

I therefore can't produce a valid latex table when the dependent variable has an underscore.

### PR DESCRIPTION
**Pull request by bashtage**
Escale string when output is LaTeX

closes #5297

- [X] closes #5297
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message).

### CODE REVIEW

File: statsmodels/iolib/tests/test_summary.py [hunk 1]
```diff
@@ -3,6 +3,24 @@
 
 '''
 from __future__ import print_function
+
+import numpy as np  # noqa: F401
```
Comment (jbrockmendel) [inline statsmodels/iolib/tests/test_summary.py]: Why is this needed?

Comment (bashtage) [reply statsmodels/iolib/tests/test_summary.py]: It is used in the formula below.

Comment (bashtage) [reply statsmodels/iolib/tests/test_summary.py]: which is evaluated in the callers frame.
